src/main/java/frc/robot/subsystems/chassis/Chassis.java

```diff
@@ -56,45 +56,61 @@ public class Chassis extends SubsystemBase {
      */
     public Chassis() {
         field = new Field2d();
-        gyro = new PigeonIMU(SwerveConstants.GYRO_ID);
+        gyro = new PigeonIMU(ChassisConstants.GYRO_ID);
         modules = new SwerveModule[] {
                 new SwerveModule(SwerveModuleConstants.FRONT_LEFT),
                 new SwerveModule(SwerveModuleConstants.FRONT_RIGHT),
                 new SwerveModule(SwerveModuleConstants.BACK_LEFT),
                 new SwerveModule(SwerveModuleConstants.BACK_RIGHT)
         };
-        angleController = new PIDController(SwerveConstants.TELEOP_ROTATION_KP,
-                SwerveConstants.TELEOP_ROTATION_KI, 0);
+        angleController = new PIDController(ChassisConstants.TELEOP_ROTATION_KP,
+                ChassisConstants.TELEOP_ROTATION_KI, 0);
         angleController.enableContinuousInput(0, 2 * Math.PI);
-        angleController.setTolerance(SwerveConstants.ANGLE_TOLERANCE);
-        poseEstimator = new SwerveDrivePoseEstimator(SwerveConstants.KINEMATICS, getGyroRotation(),
+        angleController.setTolerance(ChassisConstants.ANGLE_TOLERANCE);
+        poseEstimator = new SwerveDrivePoseEstimator(ChassisConstants.KINEMATICS, getGyroRotation(),
                 getModulePositions(), new Pose2d(0, 0, getGyroRotation()));
         isBreak = true;
 
-        SmartDashboard.putData(this);
-
         startPitch = gyro.getPitch();
         startRoll = gyro.getRoll();
+
+        setupVisionListener();
+
+        SmartDashboard.putData(this);
```

Comment:
I'd prefer that this would happen when you create the instance (for less magic code and more customizable options).

Reply:
it's in the constructor wdym